src/rabbit_control_main.erl

```diff
@@ -285,17 +285,20 @@ shutdown_node_and_wait_pid_to_stop(Node, Pid, Inform) ->
               "RabbitMQ node ~p running at PID ~s successfully shut down",
               [Node, Pid]),
             Res;
-        {error, Err} ->
-            {error, {error_during_shutdown, Err}};
-        _  ->
-            Res
+        {badrpc, {'EXIT', RpcErr}} ->
+            {error, {error_during_shutdown, RpcErr}};
+        Error ->
+            {error, {error_during_shutdown, Error}}
     end.
 
 action(shutdown, Node, [], _Opts, Inform) ->
     case rpc:call(Node, os, getpid, []) of
         Pid when is_list(Pid) ->
             shutdown_node_and_wait_pid_to_stop(Node, Pid, Inform);
-        Error -> Error
+        {badrpc, {'EXIT', RpcErr}} ->
+            {error, {error_during_shutdown, RpcErr}};
+        Error ->
```

Comment:
We don't want to wrap all possible errors into `error_during_shutdown`, only those that we are absolutely sure related to `rabbit:stop_and_halt/0` execution.